xarray/core/dataarray.py

```diff
@@ -951,6 +951,52 @@ def interp(self, coords=None, method='linear', assume_sorted=False,
             **coords_kwargs)
         return self._from_temp_dataset(ds)
 
+    def interp_like(self, other, method='linear', assume_sorted=False,
+                    kwargs={}):
+        """Interpolate this object onto the coordinates of another object,
+        filling out of range values with NaN.
+
```

Comment:
Can we add a note that if there are coordinates with object dtype, these are reindexed instead, not interpolated?